Add integration tests for ironsftp

BobWall23 opened the pull request:
Changed make so by default it only builds ironsftp. Added iron-tests target. This runs the ironcore unit and integration tests.
Created integration tests for ironsftp.
Cleaned up a bug in iron_generate_keys where it didn't create the pubkeyidx directory.

BobWall23 committed:
Add integration tests for ironsftp

Changed make so by default it only builds ironsftp. Added iron-tests
target. This runs the ironcore unit and integration tests.
Created integration tests for ironsftp.
Cleaned up a bug in iron_generate_keys where it didn't create the
pubkeyidx directory.

BobWall23 committed:
Add target that got lost in the shuffle

zmre commented on iron-gpg.h:
```diff
@@ -38,6 +38,7 @@
 
 
 extern int  iron_initialize(void);
+extern void iron_set_user(const char * path);
```
`iron_set_user_path`?

BobWall23 replied:
It sets the user login and the path.

zmre commented on iron/gpg.c:
```diff
@@ -51,17 +51,73 @@
+ *  @param path The base path for the user
+ */
+static void
+iron_set_user_dir(const char * path)
```
I'm very confused. The comments for `iron_set_user_dir` and `iron_set_user` are identical.  How do these functions actually differ? What's the point in having two?

BobWall23 replied:
Cut and paste error. `iron_set_user` sets the login, then calls `iron_set_user_path` to set the paths. I'll fix the comment.

zmre commented on regress/iron-test-exec.sh:
```diff
@@ -0,0 +1,419 @@
+
+# Unbreak GNU head(1)
+_POSIX2_VERSION=199209
+export _POSIX2_VERSION
```
`head` requires some bizarre magic version number?

BobWall23 replied:
No clue. Copied the file from `test-exec.sh` and pared it down. I didn't know about this so I figured no point in removing it.

zmre commented on regress/iron-test-exec.sh:
```diff
@@ -0,0 +1,419 @@
+SSHD=sshd
+SSHKEYGEN=ssh-keygen
+IRONSFTP=ironsftp
+SFTPSERVER=/usr/libexec/sftp-server
```
i suspect this hard coded path may create issues

BobWall23 replied:
I'll just make `sftp-server` and use that one.

zmre commented on regress/iron-test-exec.sh:
```diff
@@ -0,0 +1,419 @@
+		SSHKEYGEN="$VG --log-file=${VG_LOG}ssh-keygen.%p $SSHKEYGEN"
+		SFTP="$VG --log-file=${VG_LOG}sftp.%p ${SFTP}"
+		SCP="$VG --log-file=${VG_LOG}scp.%p $SCP"
+		cat > $OBJ/valgrind-sftp-server.sh << EOF
```
Presumably this isn't code you added despite what diff says, since I would expect to see ironsftp on the valgrind list.

BobWall23 replied:
The whole file is new - copied from `test-exec.sh`. I haven't had time to work out valgrind yet, so I left it alone. I have a separate issue for valgrind.

BobWall23 committed:
Code review updates

Fixed comment for iron_set_user_dir.
Build sftp-server and use that for integration tests, to avoid path
problems finding the system one.
Trimmed down valgrind test list.

BobWall23 committed:
Fix path to .ironpubkey

In case where test dir was specified, but keys were already generated,
it was uploading the user's .ironpubkey file to the wrong location.
Fixed so the file gets written to the test dir.

BobWall23 committed:
Fix enc-dec-file

Updated the enc-dec-file test utility to allow caller to specify the
path to the directory containing keys. (Change similar to ironsftp -
optional -T argument that passes the path to iron_set_user()).

BobWall23 committed:
Fixed missing files in clean targets

When I moved ironsftp from TARGETS to IRON_TARGETS, forgot to add
IRON_TARGETS to the "clean:" and "distclean:" targets in the Makefile.

BobWall23 committed:
Add integration tests to build

BobWall23 committed:
Split out slow tests

Created new make target iron-slow-tests. This will run the encode/decode
test on a 2G file, run 10 times as many random 32k file encode/decodes,
and run the ironsftp tests wit the 5 and 1000 byte buffer sizes and the
1 and 2 request windows. The regular iron-tests runs significantly
faster now.

BobWall23 committed:
Merge remote-tracking branch 'upstream/master'
